src/Microsoft.Framework.PackageManager/Utils/LockFileUtils.cs

```diff
@@ -129,5 +146,97 @@ private static List<string> GetPackageAssemblies(IPackage package, FrameworkName
             return results;
         }
 
+        private static bool IsAssemblyServiceable(string assemblyPath)
+        {
+            using (var stream = File.OpenRead(assemblyPath))
+            using (var peReader = new PEReader(stream))
+            {
+                var mdReader = peReader.GetMetadataReader();
+                var attrs = mdReader.GetAssemblyDefinition().GetCustomAttributes()
+                    .Select(ah => mdReader.GetCustomAttribute(ah));
+
+                foreach (var attr in attrs)
+                {
+                    var ctorHandle = attr.Constructor;
+                    if (ctorHandle.Kind != HandleKind.MemberReference)
+                    {
+                        continue;
+                    }
+
+                    var container = mdReader.GetMemberReference((MemberReferenceHandle)ctorHandle).Parent;
+                    var name = mdReader.GetTypeReference((TypeReferenceHandle)container).Name;
+                    if (!string.Equals(mdReader.GetString(name), "AssemblyMetadataAttribute"))
+                    {
+                        continue;
+                    }
+
+                    var arguments = GetFixedStringArguments(mdReader, attr);
+                    if (arguments.Count == 2 &&
+                        string.Equals(arguments[0], "Serviceable", StringComparison.OrdinalIgnoreCase) &&
```

Comment:
Actually that's the key of metadata, not a type name:

![serviceable](https://cloud.githubusercontent.com/assets/1383883/6755954/e241701e-cee3-11e4-90f9-e07d71d47d0e.png)

@glennc , is the metadata key and value case-sensitive? Do we accept
```
[assembly: AssemblyMetadata("serviceable", "true")]
```
in addition to
```
[assembly: AssemblyMetadata("Serviceable", "True")]
```